RIPS/rip-9996.md

```diff
@@ -0,0 +1,137 @@
+---
+rip: 0
+title: An RIP-7560 two-dimensional nonce transaction subtype
+description: An RIP-7560 transaction subtype that allows Smart Contract Accounts to define their own transaction sequencing
+author: Vitalik Buterin (@vbuterin), Yoav Weiss (@yoavw), Alex Forshtat (@forshtat), Dror Tirosh (@drortirosh), Shahaf Nacson (@shahafn)
+discussions-to:
+status: Draft
+type: Standards Track
+category: Core
+created: 2023-09-01
+requires: 7560
+---
+
+## Abstract
+
+An RIP-7560 transaction subtype that replaces the existing nonce-based sequencing mechanism
+of Ethereum with an on-chain pre-deployed contract in order to provide Smart Contract Accounts
+with the flexibility they need to support some advanced use cases.
+
+## Motivation
+
+The traditional nonce-based system provides guarantees of transaction ordering and replay protection,
+but does so at the expanse of configurability.
+Accounts are not able to express their intentions like "allow these three transactions to be mined in
+whatever order" or "these two sets of transactions must have separate sequential nonces".
+
+However, with Smart Contract Accounts this creates a bottleneck for some use-cases.
+For example, Smart Contract Accounts designed to be operated by multiple participants simultaneously,
+will require these participants to coordinate their transactions to avoid invalidating each other.
+
+Another example when this can also be a limitation is a case where there are separate execution flows.
+A configuration change may require multiple participants to co-sign a transaction but a regular operation does not.
+With sequential nonces, all operations will have to be halted until the configuration change is executed.
+
+Having a more agile ordering and replay protection system will significantly improve the user experience with RIP-7560.
+
```

Comment:
Another motivation is that it's required for ERC-4337 compatibility.  All 4337 accounts already use NonceManager.

Reply:
Added.